lib/mapper.js

```diff
@@ -47,6 +48,129 @@ exports.track = function(track){
 };
 
 /**
+ * Map Completed Order
+ *
+ * Docs: http://docs.klaviyo.com/article/60-ecommerce-integrating-a-custom-cart
+ *
+ * Must call Top level request for "Placed Order" and separate request for each item in products array
+ * @param {Track} track
+ * @return {Oject}
+ * @api private
+ */
+
+exports.completedOrder = function(track, settings){
+  var products = track.products();
+  var categories = formatCategories(products);
+  var productNames = formatNames(products);
+  var items = formatItems(products);
+
+  // TODO return two payloads
+  var payloads = {};
+
+  // Placed Order payload
+  payloads.order = {
+    token: settings.apiKey,
+    event: 'Placed Order',
+    time: time(track.timestamp())
+  };
+
+  payloads.order.customer_properties = {
+    $id: track.userId() || track.sessionId()
+  };
+
+  payloads.order.properties = {
+    $event_id: track.orderId(),
+    $value: track.revenue(),
+    Categories: categories,
+    'Item Names': productNames,
+    'Items': items
+  };
+
+  // Ordered Product payloads
+  payloads.products = products.map(function(item){
+    var product = new Track({ properties: item });
+    var payload = {
+      token: settings.apiKey,
+      event: 'Ordered Product',
+      time: time(track.timestamp())
+    };
+    payload.customer_properties = {
+      $id: track.userId() || track.sessionId()
+    };
+    payload.properties = {
+      $event_id: product.id() || track.orderId() + '_' + product.sku(),
+      $value: product.price(),
+      Name: product.name(),
+      Quantity: product.quantity(),
+      "Product Categories": [product.category()]
+    };
+    if (product.proxy('properties.productUrl')) payload.properties['Product URL'] = product.proxy('properties.productUrl');
+    if (product.proxy('properties.imageUrl')) payload.properties['Image URL'] = product.proxy('properties.imageUrl');
+
+    return payload;
+  });
+
+  return payloads;
+}
+
+/**
+ * Format categories of products.
+ *
+ * @param {Array} products
+ * @return {Array}
+ * @api private
+ */
+
+function formatCategories(products){
```

Comment:
No fix necessary here—but generally speaking when you write a function that is in the plural, you should ask yourself if you can write a singular version. What that usually results in is either: 1. you figure out the plural version is unnecessary (in this case, it's just a call to `map`), and/or 2: you get both a singular and plural function which are much more decoupled